core-bundle/src/Filesystem/FilesystemItem.php

```diff
@@ -174,6 +210,56 @@ public function getUuid(): Uuid|null
         return $this->getExtraMetadata()['uuid'] ?? null;
     }
 
+    public function getSchemaOrgData(): array
+    {
+        $fileIdentifier = $this->getPath();
+
+        if ($this->getUuid()) {
+            $fileIdentifier = '#/schema/file/'.$this->getUuid();
+        }
+
+        $type = match (true) {
+            $this->isVideo() => 'VideoObject',
+            $this->isAudio() => 'AudioObject',
+            $this->isImage() => 'ImageObject',
+            $this->isPdf() => 'DigitalDocument',
+            $this->isSpreadsheet() => 'SpreadsheetDigitalDocument',
+            default => 'MediaObject',
+        };
+
+        $jsonLd = [
+            '@type' => $type,
+            'identifier' => $fileIdentifier,
+            'contentUrl' => $this->getPath(),
+            'encodingFormat' => $this->getMimeType(),
+        ];
+
+        if ($this->getMetaData()) {
+            $jsonLd = [...$this->getMetaData()->getSchemaOrgData($type), ...$jsonLd];
+        }
+
+        ksort($jsonLd);
+
+        return $jsonLd;
+    }
+
+    private function getMetaData(): Metadata|null
+    {
+        $metadataBag = $this->getExtraMetadata()['metadata'] ?? null;
+
+        if (!$metadataBag instanceof MetadataBag) {
+            return null;
+        }
+
+        $metadata = $metadataBag->getDefault();
+
+        if (!$metadata instanceof Metadata) {
+            return null;
+        }
+
+        return $metadata;
```

Comment:
```suggestion
        return ($this->getExtraMetadata() ?? null)?->getDefault();
```

Reply:
That wouldn't work because it's missing the check for the nested `metadata` array key.

Reply:
Oops, that was missing accidentally - I updated the suggestion.